operator/pkg/component/component/component.go

```diff
@@ -180,7 +185,72 @@ func (c *PilotComponent) RenderManifest() (string, error) {
 	if !c.started {
 		return "", fmt.Errorf("component %s not started in RenderManifest", c.ComponentName())
 	}
-	return renderManifest(c.CommonComponentFields)
+	baseYAML, err := renderManifest(c.CommonComponentFields)
+	if err != nil {
+		return "", err
+	}
+
+	return c.overlayMeshConfig(baseYAML)
+}
+
+func (c *PilotComponent) overlayMeshConfig(baseYAML string) (string, error) {
+	if c.CommonComponentFields.InstallSpec.MeshConfig == nil {
+		return baseYAML, nil
+	}
+
+	// Overlay MeshConfig onto the istio configmap
+	baseObjs, err := object.ParseK8sObjectsFromYAMLManifest(baseYAML)
+	if err != nil {
+		return "", err
+	}
+
+	for _, obj := range baseObjs {
+		if obj.Kind != "ConfigMap" {
+			continue
+		}
+
+		// The configmap is technically istio{{version}}
```

Comment:
nit: combine the two if statements to make more compact. also, i'd recommend creating a helper function func isMeshConfigMap(o object) (bool, error) below to abstract this detail a bit.